cluster/manager.go

```diff
@@ -106,15 +107,15 @@ func (m *Manager) getPrometheusTimer(provider, location, status string, orgId ui
 	return prometheus.NewTimer(m.statusChangeDurationMetric.WithLabelValues(provider, location, status, "", "")), nil
 }
 
-func (m *Manager) GetKubeProxy(apiProxyPrefix string, commonCluster CommonCluster) (*KubeAPIProxy, error) {
+func (m *Manager) GetKubeProxy(requestUrl *url.URL, apiProxyPrefix string, commonCluster CommonCluster) (*KubeAPIProxy, error) {
 	// Currently we do not lock this transaction of getting and optionally creating a KubeAPIProxy.
 	// The worst thing that could happen is that for a short period (a Go GC period) there will be
 	// an extra KubeAPIProxy object in memory, but we can keep this method lock-free I think this is a good trade-off.
 	kubeProxy, found := m.kubeProxyCache.Get(commonCluster.GetUID())
 	if !found {
 		var err error
 
-		kubeProxy, err = NewKubeAPIProxy(apiProxyPrefix, commonCluster, defaultProxyExpirationMinutes*time.Minute)
+		kubeProxy, err = NewKubeAPIProxy(requestUrl, apiProxyPrefix, commonCluster, defaultProxyExpirationMinutes*time.Minute)
```

Comment:
The proxy is cached globally for a cluster UID.

Reply:
Yes the request url is the same for each cluster, so it's not a problem that we have the cache by clusterId.

Reply:
Well, that's not necessarily true (eg. if I add a random query to the API, the URL won't be the same. As far as I can see only the host and the scheme is used at the moment. Maybe it's worth a comment somewhere saying that relying on other values might cause caching issues.